core/src/main/scala/org/apache/spark/storage/DiskStore.scala

```diff
@@ -152,7 +153,7 @@ private class DiskBlockData(
     file: File,
     blockSize: Long) extends BlockData {
 
-  override def toInputStream(): InputStream = new FileInputStream(file)
+  override def toInputStream(): InputStream = new NioBufferedFileInputStream(file)
```

Comment:
IIUC, the returned `InputStream` will be deserialized in `BlockManger`, And deserializer will copy the data from direct memory to on-heap memory, otherwise how do we visit POJO?

So unless if we purely manipulate binary data, otherwise we have to copy the data to on-heap. Please correct me if I'm wrong.

Besides, I think this is not the hotspot, so memory copying should not bring in big overhead.

Reply:
Hi @jerryshao, thanks for reviewing. This is inspired by #15408. 
> the returned `InputStream` will be deserialized in `BlockManger`

This is not entirely correct. Sometimes we don't need deserialized, such as network transmission. And also, this does not add extra work to deserialization, but reduces the effort of network-like delivery.

Reply:
IIUC for network (netty) transmission, it uses zero copy sendFile, which is another path (`toNetty`). Here `toInputStream` seems only used by `BlockManager` from code.

Also this is not the critical path, reducing memory copy will not increase the performance a lot. Besides, due to the complexity of the code stack, we can guarantee there's no memory-copy in the whole code chain, so I doubt about the final improvements of changes.

Reply:
>IIUC for network (netty) transmission, it uses zero copy sendFile, which is another path (toNetty). 

Thanks for explaining, I did not notice this before.